packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts

```diff
@@ -0,0 +1,279 @@
+import {
+    BacktraceBreadcrumbs,
+    BreadcrumbLogLevel,
+    BreadcrumbType,
+    BreadcrumbsEventSubscriber,
+} from '@backtrace/sdk-core';
+import { BrowserWindow, Event, RenderProcessGoneDetails, WebContentsWillFrameNavigateEventParams, app } from 'electron';
+import { point } from '../attributes/helpers/attributes';
+
+export class WindowEventSubscriber implements BreadcrumbsEventSubscriber {
+    private readonly _toDispose: Array<() => void> = [];
+
+    public start(breadcrumbs: BacktraceBreadcrumbs): void {
+        app.on('browser-window-created', (_, window) => this.registerWindow(window, breadcrumbs));
+
+        for (const window of BrowserWindow.getAllWindows()) {
+            this.registerWindow(window, breadcrumbs);
+        }
+    }
+
+    public dispose(): void {
+        for (const dispose of this._toDispose) {
+            dispose();
+        }
+    }
+
+    private registerWindow(window: BrowserWindow, breadcrumbs: BacktraceBreadcrumbs) {
+        const disposeWindowEvents = this.listenForWindowEvents(window, breadcrumbs);
+        const disposeWebContentsEvents = this.listenOnWebContentsEvents(window, breadcrumbs);
+        const dispose = () => {
+            disposeWindowEvents();
+            disposeWebContentsEvents();
+        };
+
+        this._toDispose.push(dispose);
+        window.once('closed', () => {
+            dispose();
+            this._toDispose.splice(this._toDispose.indexOf(dispose), 1);
+        });
+    }
+
+    private listenForWindowEvents(window: BrowserWindow, breadcrumbs: BacktraceBreadcrumbs) {
+        const windowMsg = (msg: string) => `Window ${window.title} (ID: ${window.id}) ${msg}`;
+
+        const onClose = () =>
```

Comment:
in this situation, we generate two logs about the same thing. Maybe it's worth marking the "onClose" event as verbose?

Reply:
Changed.